sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -2532,26 +2500,16 @@ object GpuOverrides {
     expr[ArrayMax](
       "Returns the maximum value in the array",
       ExprChecks.unaryProject(
-        TypeSig.commonCudfTypes + TypeSig.DECIMAL_64 + TypeSig.NULL,
+        // TODO add back in STRING support once https://github.com/rapidsai/cudf/issues/9156
+        //   is fixed
+        TypeSig.commonCudfTypes + TypeSig.DECIMAL_64 + TypeSig.NULL - TypeSig.STRING,
         TypeSig.orderable,
-        TypeSig.ARRAY.nested(TypeSig.commonCudfTypes + TypeSig.DECIMAL_64 + TypeSig.NULL),
+        TypeSig.ARRAY.nested(TypeSig.commonCudfTypes + TypeSig.DECIMAL_64 + TypeSig.NULL
+            - TypeSig.STRING),
         TypeSig.ARRAY.nested(TypeSig.orderable)),
       (in, conf, p, r) => new UnaryExprMeta[ArrayMax](in, conf, p, r) {
         override def tagExprForGpu(): Unit = {
-          if (isOrContainsFloatingPoint(in.dataType)) {
-            if (!conf.isFloatAggEnabled) {
-              willNotWorkOnGpu("the GPU will aggregate floating point values in" +
-                  " parallel and the result is not always identical each time. This can cause" +
-                  " some Spark queries to produce an incorrect answer if the value is computed" +
-                  " more than once as part of the same query.  To enable this anyways set" +
-                  s" ${RapidsConf.ENABLE_FLOAT_AGG} to true.")
-            }
-            if (conf.hasNans) {
-              willNotWorkOnGpu("Max aggregation on floating point columns that can contain NaNs" +
-                  " will compute incorrect results. If it is known that there are no NaNs, set" +
-                  s" ${RapidsConf.HAS_NANS} to false.")
-            }
-          }
+          checkAndTagFloatNanAgg("Max", in.dataType, conf, this)
```

Comment:
Missing `checkAndTagFloatAgg` call here?

Reply:
Same here I fixed a bug I had before that I caught because I combined the checks and I had to think why the regular min and max aggregations didn't do the check I added for the array versions.